### PR TITLE
Add ${maven.home} to all surefire and failsafe executions 

### LINKED ISSUE #9505: QuarkusTest is downloading dependencies to default localRepository (ignores MAVEN_HOME/conf/settings.xml?)
**Describe the bug**
While trying out 1.5.0.CR1, I saw duplicate downloads: First the "regular" ones by the main Maven process but then a second time when the first `@QuarkusTest` was executed.
I then renamed `~/.m2/repository` and voilà: The main Maven process does not re-download anything but the test execution does.
It seems the test execution is not picking up my settings from `MAVEN_HOME/conf/settings.xml`.
Please note that I have multiple local repositories for isolated development and therefore I do _not_ set `localRepository` via `~/.m2/settings.xml`.

**Expected behavior**
QuarkusTests must use the same Maven settings as the main Maven process.

**Actual behavior**
- `localRepository` from `MAVEN_HOME/conf/settings.xml` is ignored
- `mirrors` from `MAVEN_HOME/conf/settings.xml` are ignored (I can see that some of the dependency downloads triggered by the QuarkusTest are performed against central)

**To Reproduce**
1. rename `~/.m2/repository` to something else
2. configure a `localRepository` in `MAVEN_HOME/conf/settings.xml`
3. build some quarkus project incl. QuarkusTest execution (you should already see duplicate downloads)
4. again rename `~/.m2/repository` to something else
5. build project again, which should now only download when executing the test

**Configuration**
n/a

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`:
    ```MINGW64_NT-10.0-18363 W4DEUMSY9003463 3.0.7-338.x86_64 2019-11-21 23:07 UTC x86_64 Msys```
 - Output of `java -version`:
    ```
    openjdk version "11.0.6" 2020-01-14
    OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
    OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
    ```
 - GraalVM version (if different from Java): n/a
 - Quarkus version or git rev: `1.4.2.Final` & `1.5.0.CR1`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): `Apache Maven 3.6.3`

**Additional context**
This is _not_ a regression caused by 1.5.0.CR1 since it also happens with 1.4.2.Final.

### PR DESCRIPTION
**Pull request by famod**
Resolves #9505

See also: https://github.com/quarkusio/quarkus/issues/9505#issuecomment-632337722

cc @geoand

### CODE REVIEW

File: docs/src/main/asciidoc/maven-tooling.adoc
Comment (geoand) [inline docs/src/main/asciidoc/maven-tooling.adoc:16]: This is a little strongly worded and would scare users who don't really understand what that configuration usually does (which is pretty much nothing by default).
How about we word it a in way that lets users know that **if** they have changed any settings in `conf/settings.xml`, then either `MAVEN_HOME` or the `maven.home` system property needs to be set?

Comment (famod) [reply docs/src/main/asciidoc/maven-tooling.adoc:16]: Ok, I'll try.

If we start mentioning `maven.home` here we should also implement @aloubyansky's idea of generating a `systemPropertyVariables` entry, no?

Comment (geoand) [reply docs/src/main/asciidoc/maven-tooling.adoc:16]: Yeah, that makes perfect sense

Comment (famod) [reply docs/src/main/asciidoc/maven-tooling.adoc:16]: So I'll take a stab at this. Once I'm done someone with a IntelliJ installation should check how it reacts to `${maven.home}` in `systemPropertyVariables`.

Comment (famod) [reply docs/src/main/asciidoc/maven-tooling.adoc:16]: Do we want to have this new `systemPropertyVariable` _everywhere_ where we already have the `LogManager` configured?
```
$ grep -Rl --exclude-dir=target '<java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>' .
./build-parent/pom.xml
./core/builder/pom.xml
./devtools/aesh/dependency-reduced-pom.xml
./devtools/aesh/pom.xml
./devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/java/pom.xml-template.ftl
./devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/kotlin/pom.xml-template.ftl
./devtools/platform-descriptor-json/src/main/resources/templates/basic-rest/scala/pom.xml-template.ftl
./docs/src/main/asciidoc/building-my-first-extension.adoc
./docs/src/main/asciidoc/getting-started-testing.adoc
./docs/src/main/asciidoc/getting-started.adoc
./docs/src/main/asciidoc/logging.adoc
./docs/src/main/asciidoc/tests-with-coverage.adoc
./extensions/amazon-lambda/maven-archetype/src/main/resources/archetype-resources/pom.xml
./extensions/amazon-lambda-http/maven-archetype/src/main/resources/archetype-resources/pom.xml
./extensions/azure-functions-http/maven-archetype/src/main/resources/archetype-resources/pom.xml
./extensions/funqy/funqy-amazon-lambda/maven-archetype/src/main/resources/archetype-resources/pom.xml
./extensions/grpc/deployment/pom.xml
./independent-projects/tools/common/src/test/resources/templates/basic-rest/java/pom.xml-template.ftl
./independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/pom.xml-template.ftl
./independent-projects/tools/common/src/test/resources/templates/basic-rest/scala/pom.xml-template.ftl
./integration-tests/bootstrap-config/application/pom.xml
./integration-tests/container-image/maven-invoker-way/src/it/container-build-docker/pom.xml
./integration-tests/container-image/maven-invoker-way/src/it/container-build-jib/pom.xml
./integration-tests/container-image/maven-invoker-way/src/it/container-build-multiple-tags-docker/pom.xml
./integration-tests/container-image/maven-invoker-way/src/it/container-build-multiple-tags-jib/pom.xml
./integration-tests/container-image/maven-invoker-way/src/it/container-image-push/pom.xml
./integration-tests/grpc-health/pom.xml
./integration-tests/grpc-interceptors/pom.xml
./integration-tests/grpc-mutual-auth/pom.xml
./integration-tests/hibernate-orm-panache/pom.xml
./integration-tests/kubernetes/maven-invoker-way/src/it/kubernetes-docker-build-and-deploy/pom.xml
./integration-tests/kubernetes/maven-invoker-way/src/it/kubernetes-jib-build-and-deploy/pom.xml
./integration-tests/kubernetes/maven-invoker-way/src/it/kubernetes-with-existing-manifest/pom.xml
./integration-tests/kubernetes/maven-invoker-way/src/it/openshift-s2i-build-and-deploy/pom.xml
./integration-tests/kubernetes/src/it/openshift-s2i-build-and-deploy/pom.xml
./integration-tests/maven/src/test/resources/projects/multimodule/runner/pom.xml
./integration-tests/maven/src/test/resources/projects/multimodule-revision-prop/runner/pom.xml
./integration-tests/maven/src/test/resources/projects/property-overrides/runner/pom.xml
./integration-tests/picocli/pom.xml
./integration-tests/spring-data-jpa/pom.xml
./integration-tests/vault/pom.xml
```
`platform-descriptor-json` is obvious but I am not sure about:
- _all_ docs
- Quarkus internal pom.xml files

Would be more consistent to do it for all, though.

Comment (aloubyansky) [reply docs/src/main/asciidoc/maven-tooling.adoc:16]: I'd say yes. `/independent-projects/tools/common/src/test/resources/templates/basic-rest/*` are formal templates, could be excluded.

Comment (famod) [reply docs/src/main/asciidoc/maven-tooling.adoc:16]: Phew, I had a quick look at some of those `pom.xml` (especially under `integration-tests`) and it all seems a bit inconsistent.
Why are so many `integration-tests`-projects redefining `maven-surefire-plugin`?
At least one of them is also passing the `LogManager` via `maven-failsafe-plugin`.

I'd say this should be cleaned up first in a separate PR. WDYT?

Comment (aloubyansky) [reply docs/src/main/asciidoc/maven-tooling.adoc:16]: It could surely be reorganized to not repeat common config. If you are up for it, sure :)

Comment (geoand) [reply docs/src/main/asciidoc/maven-tooling.adoc:16]: A cleanup is certainly welcome if you want to take up the task 😉

**Comment by famod** [comment]
I'm not done yet with the rewording of the documentation but feel free to check the rest.

Btw, I added `maven.home` to all `systemPropertyVariables` which already had the `LogManager` configured.
In some `src/test` folders there are `failsafe` configurations that do _not_ contain the `LogManager`.
Do we want to add `LogManager` and `maven.home` for these as well?

**Comment by geoand** [comment]
> In some `src/test` folders there are `failsafe` configurations that do _not_ contain the `LogManager`.
> Do we want to add `LogManager` and `maven.home` for these as well?

Can you show an example?

**Comment by famod** [comment]
> Can you show an example?

Sure:
- https://github.com/quarkusio/quarkus/blob/master/integration-tests/kubernetes/maven-invoker-way/src/it/openshift-s2i-build-and-deploy/pom.xml#L107
- https://github.com/quarkusio/quarkus/blob/master/extensions/azure-functions-http/maven-archetype/src/main/resources/archetype-resources/pom.xml#L186

There are many more.

**Comment by geoand** [comment]
I don't see any harm in adding them

**Comment by famod** [comment]
For the documentation part I'll wait until #9604 is merged.

**Comment by famod** [comment]
I'm done (I think) with the documentation.

I decided to relocate and extend the new section "Running `@QuarkusTest` from an IDE" (that has just been introduced via #9604)  from logging to getting-started-testing. I left a cross reference.

@sorin-costea maybe you want to check whether that is ok for you.

File: docs/src/main/asciidoc/maven-tooling.adoc
Comment (famod) [reply docs/src/main/asciidoc/maven-tooling.adoc]: @geoand 

> This is a little strongly worded and would scare users who don't really understand what that configuration usually does (which is pretty much nothing by default).
> How about we word it a in way that lets users know that **if** they have changed any settings in `conf/settings.xml`, then either `MAVEN_HOME` or the `maven.home` system property needs to be set?

That initial comment has just been resolved: The section is now gone. 😄

File: docs/src/main/asciidoc/logging.adoc
Comment (famod) [inline docs/src/main/asciidoc/logging.adoc:252]: I hope this link works. I wasn't abe to test this properly.

Comment (geoand) [reply docs/src/main/asciidoc/logging.adoc:252]: It should, but I'm not an adoc expert. @gastaldi perhaps?

Comment (gastaldi) [reply docs/src/main/asciidoc/logging.adoc:252]: Sorry, I'm no adoc expert either :) 

Try building the project and see the output in `docs/target/generated-docs'

Comment (famod) [reply docs/src/main/asciidoc/logging.adoc:252]: > Try building the project and see the output in `docs/target/generated-docs`

🤦 Sure, silly me. Just checked: It works!

**Comment by famod** [comment]
CI is broken: `##[error]Docker pull failed with exit code 1`

**Comment by geoand** [comment]
> CI is broken: `##[error]Docker pull failed with exit code 1`

I think `quay.io` is probably having issues again...

**Review by aloubyansky — APPROVED** [review]

**Comment by famod** [comment]
> > CI is broken: `##[error]Docker pull failed with exit code 1`
> 
> I think `quay.io` is probably having issues again...

Yeah, seems so: https://status.quay.io/

> We are still investigating the root cause and continue to work towards a solution. Multiple mitigation approaches are being evaluated. Next update in 60 minutes or as new information becomes available.
> May 28, 10:09 EDT

**Comment by famod** [comment]
I just rebased onto `master` to re-trigger CI (quay.io seems ok now).

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by famod** [comment]
Ah, it seems there is one thing missing: How does IntelliJ react to `${maven.home}`? Is it working?
Can someone with a IntelliJ installation test this please? Thanks!

**Comment by geoand** [comment]
IntelliJ does pick up the new `maven.home` property when it's set in `systemPropertyVariables` (but not when `systemProperties` is used)

**Comment by famod** [comment]
This should be good to merge, then.

**Comment by geoand** [comment]
Thanks for picking this up @famod

**Comment by famod** [comment]
I am wondering whether this should be backported (to 1.5.1)? At least one other user stubled upon this: https://github.com/quarkusio/quarkus-quickstarts/issues/586

**Comment by geoand** [comment]
Back-porting would only help for the newly created projects

**Comment by famod** [comment]
Yeah, but what about the documentation? It would take another month I guess until it becomes availabe at quarkus.io.

**Comment by geoand** [comment]
Good catch. @gsmet we don't have another avenue to getting doc updates to the website do we?

**Comment by famod** [comment]
> Good catch. @gsmet we don't have another avenue to getting doc updates to the website do we?

WDYT @gsmet?

**Pull request merged**
